fpdf/html.py

```diff
@@ -280,15 +288,48 @@ def handle_data(self, data):
                 align=self.align[0].upper(),
                 link=self.href,
             )
+        elif self.pre_formatted:  # for pre blocks
+            self.pdf.write(self.h, data)
+
+        elif (
+            self.follows_fmt_tag and not self.follows_trailing_space
+        ):  # don't trim leading whitespace if following a format tag with no trailing whitespace
+            data = self.whitespace.sub(whitespace_repl, data)
+            if trailing_space_flag:
+                self.follows_trailing_space = True
+            if self.href:
+                self.put_link(data)
+            else:
+                if self.heading_level:
+                    self.pdf.start_section(data, self.heading_level - 1)
+                LOGGER.debug(
+                    "write '%s' h=%d",
+                    self.whitespace.sub(whitespace_repl, data),
+                    self.h,
+                )
+                self.pdf.write(self.h, data)
+            self.follows_fmt_tag = False
+
         else:
-            data = data.replace("\n", " ")
+            data = self.leading_space.sub(leading_whitespace_repl, data)
+            data = self.whitespace.sub(whitespace_repl, data)
+
+            if trailing_space_flag:
```

Comment:
```suggestion
self.follows_trailing_space = trailing_space_flag
```
This single line could replace the `if` over 4 lines